src/AstNode.h

```diff
@@ -88,6 +88,12 @@ class AstNode {
         return out;
     }
 
+    std::string show() const {
```

Comment:
We would really like to keep the AST interface simpler, especially the base AstNode. In this case show is also close to a duplication of operator<< and print. It could be a toString function in AstUtils.h, but given it's only used in DebugReporter it could stay in DebugReporter.

Reply:
Technically it isn't needed at all. `toString` can work just fine, but it instantiates functions specific to node types explicitly used in the program. My use case is pretty-printing ASTs from within a debugger, which means I might be trying to pprint a node type for which there is no instantiated `toString` instance.

I've removed `AstNode::show` and added an equiv function named `pprint` to `AstUtils`. As before, we use it in DebugReporter to ensure its linked in and thus available during debugging.